### PR TITLE
Add ScreenCapturer.cs

### PR DESCRIPTION
**Pull request by amishsaxena**
Added ScreenCapturer class. Updated MakeScreenshot method's permission to public.

### CODE REVIEW

File: src/PlexShareScreenshare/Client/ScreenCapturer.cs [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+namespace PlexShareScreenshare.Client
+{
+    /// <summary>
+    /// Class contains implementation of the screen capturing using threads (tasks)
+    /// </summary>
+    internal class ScreenCapturer
+    {
+        private Queue<Bitmap> CapturedFrame;
+
+        // limits the number of frames in the queue
```
Comment (tzublue) [inline src/PlexShareScreenshare/Client/ScreenCapturer.cs]: nit - L should be capital

File: src/PlexShareScreenshare/Client/ScreenCapturer.cs [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        private CancellationTokenSource Source;
+        private CancellationToken Token;
+
+        ScreenCapturer() 
+        {
+            CapturedFrame = new Queue<Bitmap>();
+            CaptureTask = null;
+        }
+
+        // Pops and return the image from the queue
```
Comment (tzublue) [inline src/PlexShareScreenshare/Client/ScreenCapturer.cs]: Remove this comment

File: src/PlexShareScreenshare/Client/ScreenCapturer.cs [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        /// Creates the task for capturing screenshots and starts capturing
+        /// </summary>
+        public void StartCapture()
+        {
+            Source = new CancellationTokenSource();
+            Token = Source.Token;
+            
+            CaptureTask = new Task(() =>
+            {
+                Screenshot screenshot = new Screenshot();
```
Comment (tzublue) [inline src/PlexShareScreenshare/Client/ScreenCapturer.cs:69]: Try catch here, taking a screenshot might give an error

File: src/PlexShareScreenshare/Client/ScreenCapturer.cs [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+using System.Threading;
+
+namespace PlexShareScreenshare.Client
+{
+    /// <summary>
+    /// Class contains implementation of the screen capturing using threads (tasks)
+    /// </summary>
+    internal class ScreenCapturer
+    {
+        private Queue<Bitmap> CapturedFrame;
```
Comment (tzublue) [inline src/PlexShareScreenshare/Client/ScreenCapturer.cs]: _capturedFrame

File: src/PlexShareScreenshare/Client/ScreenCapturer.cs [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+
+        // limits the number of frames in the queue
+        const int MaxQueueLength = 50;
+
+        // Task to capture the screen asynchronously
+        private Task CaptureTask;
```
Comment (tzublue) [inline src/PlexShareScreenshare/Client/ScreenCapturer.cs]: _captureTask

Comment (amishsaxena) [reply src/PlexShareScreenshare/Client/ScreenCapturer.cs]: Done

File: src/PlexShareScreenshare/Client/ScreenCapturer.cs (hunk 4 above)
Comment (amishsaxena) [reply src/PlexShareScreenshare/Client/ScreenCapturer.cs]: Done

File: src/PlexShareScreenshare/Client/ScreenCapturer.cs (hunk 3 above)
Comment (amishsaxena) [reply src/PlexShareScreenshare/Client/ScreenCapturer.cs:69]: Added where MakeScreenshot() is called

File: src/PlexShareScreenshare/Client/ScreenCapturer.cs (hunk 2 above)
Comment (amishsaxena) [reply src/PlexShareScreenshare/Client/ScreenCapturer.cs]: Done

File: src/PlexShareScreenshare/Client/ScreenCapturer.cs (hunk 1 above)
Comment (amishsaxena) [reply src/PlexShareScreenshare/Client/ScreenCapturer.cs]: Done

**Pull request merged**